runtime/compiler/control/CompilationRuntime.hpp

```diff
@@ -644,6 +641,25 @@ class CompilationInfo
          }
       return success;
       }
+   // If the invocation count is 0, set it to the value indicated by newCount
+   static bool replenishInvocationCountIfExpired(J9Method *method, int32_t newCount)
+      {
+      intptr_t oldMethodExtra = (intptr_t) method->extra;
+      if ((oldMethodExtra & J9_STARTPC_NOT_TRANSLATED) == 0)
+         return false; // Do not touch compiled methods
+
+      int32_t oldCount = (int32_t)oldMethodExtra;
+      if (oldCount < 0)
+         return false; // Do not touch uncountable methods
+      oldCount >>= 1; // Eliminate the J9_STARTPC_NOT_TRANSLATED bit
+      if (oldCount != 0)
```

Comment:
Should this be `if ((oldCount & J9_INVOCATION_COUNT_MASK) != 0)`? Since there's an upper part you're addressing below.

Reply:
Now that I think of it, same for `< 0` check above too right?

Reply:
`oldCount` is a 32-bit quantity. When we did the cast `int32_t oldCount = (int32_t)oldMethodExtra;` the upper part disappeared.